waltz-ng/client/data-types/components/data-type-tree.js

```diff
@@ -0,0 +1,67 @@
+import {initialiseData, buildHierarchies} from "../../common";
+import {summerFactory} from "../../common/tally-utils";
+
+
+const bindings = {
+    dataTypes: '<',
+    tallies: '<'
+};
+
+
+const initialState = {
+    dataTypeHierarchy: []
+};
+
+
+const template = require('./data-type-tree.html');
+
+
+function prepareDataTypeTree(dataTypes, tallies) {
+
+    const dataTypesByCode = _.keyBy(dataTypes, 'code');
+
+    const enrichWithDirectCounts = (tallies, keyName) => {
+        _.each(tallies, t => {
+            const dt = dataTypesByCode[t.id];
+            if (dt) dt[keyName] = t.count;
+        });
+    };
+
+    enrichWithDirectCounts(tallies, "dataFlowCount");
```

Comment:
Ps. this has changed somewhat as i started working on the data type view
page.  It's more like the org unit version of the tree now.

On Wed, Aug 17, 2016 at 5:37 PM, David Watkins <notifications@github.com>
wrote:

> In waltz-ng/client/data-types/components/data-type-tree.js
> <https://github.com/khartec/waltz/pull/424#discussion_r75158956>:
>
> > +
> > +const template = require('./data-type-tree.html');
> > +
> > +
> > +function prepareDataTypeTree(dataTypes, tallies) {
> > +
> > +    const dataTypesByCode = _.keyBy(dataTypes, 'code');
> > +
> > +    const enrichWithDirectCounts = (tallies, keyName) => {
> > +        _.each(tallies, t => {
> > +            const dt = dataTypesByCode[t.id];
> > +            if (dt) dt[keyName] = t.count;
> > +        });
> > +    };
> > +
> > +    enrichWithDirectCounts(tallies, "dataFlowCount");
>
> Ok, understood. Will probably need to move this to a common function soon
> as we'll be summing over different hierarchies in the nr. future.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/424/files/963b72b8ac3fed06aab0d66e3fdf7e616b30456d#r75158956>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC4520yHR_mF7Fxi8xrFZxH-8Iu9_AKfks5qgzjdgaJpZM4JjQIq>
> .
>



-- 


*Kamran Saleem*
t  +44 (0) 7732 078654
e  kamran@thinkincode.co.uk
w thinkincode.co.uk